### PR TITLE
Fix overeager scrolling in TreeMap/Grid combo.  Related TreeMap cleanups

### PR DESCRIPTION
**Pull request by lbwexler**
+ Related and Unrelated tweaks

Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [X] Caught up with `develop` branch as of last change.
- [X] Added CHANGELOG entry, or determined not required.
- [X] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [X] Updated doc comments / prop-types, or determined not required.
- [X] Reviewed and tested on Mobile, or determined not required.
- [X] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

File: desktop/cmp/treemap/TreeMap.js [hunk 1]
```diff
@@ -103,24 +104,23 @@ class LocalModel extends HoistModel {
...
 
-        // Render HighChart when chartElem container ready in DOM, a relevant observable on the
-        // model API changes, and/or the source data array itself changes.
+        // Render HighChart when chartElem container ready in DOM, or dependencies updated
+        const chartDependencies = () => ([
+            this.chartRef.current,
+            model.highChartsConfig,
+            model.data,
+            XH.darkTheme
+        ]);
```
Comment (TomTirapani) [inline desktop/cmp/treemap/TreeMap.js:113]: We're missing the dependency on the `model.algorithm` here. The chart needs to be redrawn when the layout algorithm changes

**Comment by lbwexler** [comment]
I built it in to the dependency for  the model's `data` itself
intentionally.  Thought that was more straightforward, and that it is
unlikely to change.

**Comment by TomTirapani** [comment]
But, it has nothing to do with how we process the data? It's essentially part of the highcharts config that we've lifted to be a bindable. Don't see how its more straightforward to rebuild the data in order to redraw the chart, when we can just redraw the chart?

File: desktop/cmp/treemap/TreeMap.js (hunk 1 above)
Comment (lbwexler) [reply desktop/cmp/treemap/TreeMap.js:114]: restored

**Pull request merged**